pkg/test/rekor_integration/rekor_monitor_integration_test.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+)
+
+func TestTamperedCheckpoint(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	buildDir := t.TempDir()
+	binary := filepath.Join(buildDir, "rekor_monitor")
+	buildCmd := exec.Command("go", "build", "-o", binary, "github.com/sigstore/rekor-monitor/cmd/rekor_monitor")
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+	if err := buildCmd.Run(); err != nil {
+		t.Fatalf("failed to build binary: %v", err)
+	}
+
+	dataDir := t.TempDir()
+	checkpointFile := filepath.Join(dataDir, "checkpoint_log.txt")
+
+	// First run to generate initial checkpoint file
+	initCmd := exec.CommandContext(ctx, binary,
+		"--once",
+		"--file", checkpointFile,
+		"--url", "https://rekor.sigstore.dev",
+	)
+	initCmd.Stdout = os.Stdout
+	initCmd.Stderr = os.Stderr
+	err := initCmd.Run()
+	if err == nil {
+		t.Fatal("expected error on initial run due to no start index")
+	}
+
+	if _, err := os.Stat(checkpointFile); err != nil {
+		t.Fatalf("checkpoint file not created: %v", err)
+	}
+
+	// Tamper with the root hash in the checkpoint file
+	content, err := os.ReadFile(checkpointFile)
+	if err != nil {
+		t.Fatalf("failed to read checkpoint file: %v", err)
+	}
+	contentStr := string(content)
+	wasNewlineTrailing := strings.HasSuffix(contentStr, "\n")
+	contentStr = strings.TrimSuffix(contentStr, "\n")
+	wasEscapedTrailing := strings.HasSuffix(contentStr, "\\n")
+	contentStr = strings.TrimSuffix(contentStr, "\\n")
+	// Split on \\n to get log name, tree size, root hash, and signature
+	lines := strings.SplitN(contentStr, "\\n", 4)
+	if len(lines) < 3 {
+		t.Fatalf("invalid checkpoint file format: expected at least 3 lines, got %d", len(lines))
+	}
+	lines[2] = lines[2] + "tampered" // Tamper with root hash
```

Comment:
I would split this test into two separate test cases. The first one is the happy path. It consists of
- start the monitor
- check created checksum
- No errors are recorded on the monitor endpoint, but the number of monitor runs increases


The second test case:
- mock tampered checksum file
- monitor the error endpoint for errors